frontend/src/ui/Blocks/Series.tsx

```diff
@@ -166,24 +168,29 @@ const ReadySeriesBlock: React.FC<ReadyProps> = ({
     const timeMs = (event: Event) =>
         new Date(event.syncedData?.startTime ?? event.created).getTime();
 
+    const sortEvents = (a: Event, b: Event) =>
+        match(eventOrder, {
+            "NEW_TO_OLD": () => timeMs(b) - timeMs(a),
+            "OLD_TO_NEW": () => timeMs(a) - timeMs(b),
+            "A-Z": () => a.title.toLowerCase() > b.title.toLowerCase() ? 1 : -1,
+            "Z-A": () => a.title.toLowerCase() > b.title.toLowerCase() ? -1 : 1,
+        }, unreachable);
+
     const sortedEvents = [...events];
     sortedEvents.sort((a, b) => {
         // Sort all live events before non-live events.
         if (a.isLive !== b.isLive) {
             return +b.isLive - +a.isLive;
         }
 
-        return match(eventOrder, {
-            "NEW_TO_OLD": () => timeMs(b) - timeMs(a),
-            "OLD_TO_NEW": () => timeMs(a) - timeMs(b),
-        }, unreachable);
+        return sortEvents(a, b);
     });
 
     // If there is only one upcoming event, it doesn't need an extra box or ordering.
     if (upcomingLiveEvents.length === 1) {
         sortedEvents.unshift(upcomingLiveEvents[0]);
     } else {
-        upcomingLiveEvents.sort((a, b) => timeMs(a) - timeMs(b));
+        upcomingLiveEvents.sort((a, b) => sortEvents(a, b));
```

Comment:
We used reverse order for the time-based order before so that the "soonest" upcoming events come first. I would probably keep the behavior. Maybe just add a parameter `reverseTime: boolean` to `sortEvents`. That means that if the user chooses "Oldest first" then the upcoming videos will be ordered "soonest latest"... which I think is fine.

Reply:
Argh, right. I knew there surely was a good reason we did this. Now that you mention it, I also remember talking about this.

So if I understand you correctly, we would allow alphabetical ordering for upcoming events, but otherwise always show soonest first, right? Yeah, I think this makes sense, as "new to old" and "old to new" don't really mean anything in relation to upcoming events. Though I wonder if there still could be any merit or use case for ordering these events latest to soonest. I mean I guess not but hmm.

Reply:
I'm suggesting:

- A-Z: order normal and upcoming events from a to z
- Z-A: order normal and upcoming events from z to a
- Newest first: order normal videos newest first, upcoming videos "soonest first"
- Oldest first: order normal videos oldest first, upcoming videos "soonest last"